### PR TITLE
gitlab_*: remove deprecated functionalities

### LINKED ISSUE #61898: gitlab_user contains deprecated call to be removed in 2.10
##### SUMMARY
gitlab_user contains call to Display.deprecated or AnsibleModule.deprecate and is scheduled for removal

```
lib/ansible/modules/source_control/gitlab_user.py:414:4: ansible-deprecated-version: Deprecated version ('2.10') found in call to Display.deprecated or AnsibleModule.deprecate
```

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
```
lib/ansible/modules/source_control/gitlab_user.py
```

##### ANSIBLE VERSION
```
2.10
```

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### STEPS TO REPRODUCE
N/A

##### EXPECTED RESULTS
N/A

##### ACTUAL RESULTS
N/A

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
* Documentation update
* Remove deprecated functionalities

Fixes: #61898 #61897 #61896 #61895 #61894 #61893

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME
lib/ansible/modules/source_control/gitlab_deploy_key.py
lib/ansible/modules/source_control/gitlab_group.py
lib/ansible/modules/source_control/gitlab_hook.py
lib/ansible/modules/source_control/gitlab_project.py
lib/ansible/modules/source_control/gitlab_runner.py
lib/ansible/modules/source_control/gitlab_user.py

### CODE REVIEW

**Review by gundalow — COMMENTED** [review]
Looks good. Do we need to put anything in Porting guide?

**Comment by Lunik** [comment]
This PR is going to have issue with #60425 (if merged)

**Comment by gundalow** [comment]
@Lunik Good spot, thanks for mentioning that. What do you think is the best way to deal with this?

**Comment by Akasurde** [comment]
Aren't this option supposed to be removed in `2.10` release ?

**Comment by Lunik** [comment]
#60425 should be merge in 2.9 because peoples are reporting issues using GitLab's modules (ex : #59976)
If not, those parameters will be an issue during the 2.9 life cycle.
However #61912 will still be applicated for 2.10

**Comment by Shaps** [comment]
I'd agree with @Lunik to merge #60425 first so we have better warnings for the 2.9 lifetime, then merge this.

**Comment by Shaps** [comment]
@Akasurde now that #60425 has been merged, can you rebase this so we can move it forward?

**Comment by Akasurde** [comment]
@Shaps Thanks for the heads up. I rebased the branch, please check and let me know. Thanks,

File: lib/ansible/modules/source_control/gitlab_runner.py [hunk 1]
```diff
@@ -311,7 +294,6 @@ def main():
     module = AnsibleModule(
         argument_spec=argument_spec,
         mutually_exclusive=[
-            ['api_url', 'url'],
             ['api_username', 'api_token'],
             ['api_password', 'api_token'],
         ],
```
Comment (Lunik) [inline lib/ansible/modules/source_control/gitlab_runner.py]: Remove `['api_url', 'url']` from `required_one_of`
Remove `['login_user', 'login_password']` from `required_together`

**Review by Lunik — COMMENTED** [review]
At first look, it seems good.

File: lib/ansible/modules/source_control/gitlab_runner.py (hunk 1 above)
Comment (Lunik) [inline lib/ansible/modules/source_control/gitlab_runner.py:299]: Remove `['api_url', 'url']` from `required_one_of`
Remove `['login_user', 'login_password']` from `required_together`

**Review by Lunik — CHANGES_REQUESTED** [review]
Those line are useless

File: lib/ansible/modules/source_control/gitlab_runner.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/source_control/gitlab_runner.py:299]: @Lunik Done. Can you please take a look ? Thanks for the review.

Comment (Lunik) [reply lib/ansible/modules/source_control/gitlab_runner.py:299]: Look's good. great work

**Comment by Lunik** [comment]
This issue require  #63690 to be merged first for backporting in `2.8` and `2.9`

**Comment by Akasurde** [comment]
@Lunik  Can you re-review this ? Thanks.

**Comment by Lunik** [comment]
shipit

**Comment by Akasurde** [comment]
@Lunik @Shaps @gundalow Thanks for the review.

**Pull request merged**